rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -280,6 +307,69 @@ def query_current_price(
         )
         return price, False
 
+    def _call_methods(
+            self,
+            pool_contract: EvmContract,
+            methods: list[str],
+            block_identifier: BlockIdentifier,
+            timestamp: Timestamp | None,
```

Comment:
why do you provide both block_identifier and timestamp? It should be stated in the docstring what those mean and what is the flow for when one is missing. I don't totally get it

Reply:
See [this comment](https://github.com/rotki/rotki/pull/8504#discussion_r1750877812)